src/net/sourceforge/kolmafia/maximizer/LetterBonus.java

```diff
@@ -0,0 +1,40 @@
+package net.sourceforge.kolmafia.maximizer;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.sourceforge.kolmafia.AdventureResult;
+
+public class LetterBonus {
+  static final Pattern NUMBER_PATTERN = Pattern.compile("[0-9]");
+
+  static double letterBonus(AdventureResult item) {
+    if (item == null) return 0;
+    return item.getName().length();
+  }
+
+  static double letterBonus(AdventureResult item, String letter) {
+    if (item == null) return 0;
+
+    Pattern letterPattern = Pattern.compile(letter, Pattern.CASE_INSENSITIVE);
+
+    int letters = 0;
+    Matcher matcher = letterPattern.matcher(item.getName());
+    while (matcher.find()) {
+      letters++;
+    }
+
+    return letters;
```

Comment:
I think since Java 9 this can be.

```suggestion
    return letterPattern.matchers(item.getName()).results().count();
```